parquet/src/main/scala/monix/connect/parquet/ParquetSubscriberT.scala

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2020-2020 by The Monix Connect Project Developers.
+ * See the project homepage at: https://connect.monix.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package monix.connect.parquet
+
+import monix.eval.Task
+import monix.execution.cancelables.AssignableCancelable
+import monix.execution.internal.Platform
+import monix.execution.{Ack, Callback, Scheduler}
+import monix.reactive.Consumer
+import monix.reactive.observers.Subscriber
+import org.apache.parquet.hadoop.ParquetWriter
+
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+
+/**
+  * A sink that writes each element of [[T]] passed, into a single parquet file.
+  *
+  * @param parquetWriter The apache hadoop generic implementation of a parquet writer.
+  * @tparam T Represents the type of the elements that will be written into the parquet file.
+  */
+private[parquet] class ParquetSubscriberT[T](parquetWriter: Task[ParquetWriter[T]]) extends Consumer[T, Long] {
+
+  def createSubscriber(callback: Callback[Throwable, Long], s: Scheduler): (Subscriber[T], AssignableCancelable) = {
+    val out = new Subscriber[T] {
+      override implicit val scheduler: Scheduler = s
+
+      //the number of parquet files that has been written that is returned as materialized value
+      private[this] var nElements: Long = 0
+      private[this] var memoizedWriter = parquetWriter.memoize
+
+      // Protects from the situation where last onNext throws, we call onError and then
+      // upstream calls onError or onComplete again
+      private[this] var isDone = false
+
+      override def onNext(record: T): Future[Ack] = {
+        memoizedWriter.map { parquetWriter =>
+          try {
+            parquetWriter.write(record)
+            nElements = nElements + 1
+            Ack.Continue
+          } catch {
+            case ex if NonFatal(ex) =>
+              onError(ex)
+              Ack.Stop
+          }
+        }.runToFuture(scheduler)
```

Comment:
running task to Future per each event seems to be fishy :/

I wish `Consumer` wouldn't require `createSubscriber` implementation because `Observable[A] => Task[B]` would be very simple to implement.

I am not sure what to do about it, I'd actually consider taking `() => ParquetWriter[T]` instead of a `Task` but idk, maybe `runToFuture` is not so bad in practice but I don't know without benchmarks

Reply:
thanks for the review, will take your suggestions and change it to take a `Coeval` instead,
furthermore, will try creating a benchmarks sub-module to compare them :)

Reply:
The benchmark results indicates that _unsafe_ method performs better than the `safe`, however it is not like this (although the error seems too be reliable, will run them again to see if the error decreases). 

**Writer**
ParquetWriterBenchmark.fromCoeval  thrpt    4  295.987 ± 269.596  ops/s
ParquetWriterBenchmark.fromTask    thrpt    4  236.013 ± 342.709  ops/s
ParquetWriterBenchmark.unsafe      thrpt    4  403.379 ± 144.174  ops/s
**Reader**
Benchmark                         Mode  Cnt     Score      Error  Units
ParquetReaderBenchmark.fromTask  thrpt    3  7414.555 ±  913.722  ops/s
ParquetReaderBenchmark.unsafe    thrpt    3  7275.114 ± 2693.346  ops/s

Reply:
What do you mean by `however it is not like this`?

Reply:
Sorry, I meant on the `reader` benchmark